cdap-api/src/main/java/co/cask/cdap/api/annotation/Capability.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates capabilities of a plugin.</p>
+ * <p>Pre-defined system capabilities are defined in {@link Capabilities}.</p>
+ * <p>Plugins can also be annotated with custom value to specify capability with business rules.
+ * The capability value is case insensitive.</p>
+ * <p>If a plugin does not specify any capability or the {@link #value()} array is empty then the default capability
```

Comment:
thinking about this some more, it means a plugin annotated with just 'pii' can't run anywhere. It might be better to define it such that if no system defined capability is specified, it gets all system defined capabilities. So if a plugin is just tagged with 'pii', it can be run natively or remotely.

Reply:
That was the idea only if system defined capability is present the it overrides the default. The presence or absence of user defined capability does not affect override system defined. I will add that in documentation.